pkg/controller/job/job_controller.go

```diff
@@ -628,37 +631,60 @@ func (jm *Controller) syncOrphanPod(ctx context.Context, key string) error {
 	}()
 
 	ns, name, err := cache.SplitMetaNamespaceKey(key)
```

Comment:
>have another queue just for label selectors
 
 having another queue would make it clean but doesn't it mean another `x` number of go routines for this purpose because generally we depend on `concurrent-job-sync` ?
 Given its a rare scenario, I think it would be an overkill.
 
 >have some type prefix (key:, or selector:) and first parse the prefix
 
 
 
 We can do this if we want to be more explicit about the differentiation. In this if we want to pass the `ns` however for `selectors`, we might have to little bit more than just `key:, or selector:` ?  i.e for `key` it will be `key:ns/name` for `selector` it will be something like `selector:<ns>:<selector` . May be something similar you are also suggesting in option 3 above ? This way we can also use `ns` when looking up in pod store [here](https://github.com/kubernetes/kubernetes/pull/127378/files#diff-a7577a7dede7864ff38c631319033714fdd0bed91108d976282e1099507e6ff0R654-R655) 
 
 
 
 Let me know, once we agree on this direction, i will make those specific changes.

Reply:
A prefix sounds like enough.

Also note that a queue can be made of any other comparable type https://github.com/kubernetes/kubernetes/blob/3ab0c84bacd059a5b5995d72706c63b46ed24c0b/staging/src/k8s.io/client-go/util/workqueue/rate_limiting_queue.go#L93

so you could have something like this:

```
type orphanPodKey struct {
  kind string
  value string
}
```